transform/mattermost-analytics/models/intermediate/product/nps/int_daily_server_nps_score.sql

```diff
@@ -0,0 +1,60 @@
+WITH base_cte AS (
+    SELECT
+        event_date,
+        server_id,
+        server_version,
+        user_role,
+        COUNT(DISTINCT CASE WHEN nps_score.score > 8 THEN nps_score.user_id ELSE NULL END) AS promoters,
+        COUNT(DISTINCT CASE WHEN nps_score.score < 7 THEN nps_score.user_id ELSE NULL END) AS detractors,
+        COUNT(DISTINCT CASE WHEN nps_score.score > 6 AND nps_score.score < 9 THEN nps_score.user_id ELSE NULL END) AS passives,
+        COUNT(DISTINCT user_id) AS nps_users
+    FROM
+          {{ ref('int_nps_score') }} nps_score
+    GROUP BY
+        event_date,
+        server_id,
+        server_version,
+        user_role
+    ORDER BY
+        event_date DESC
+), tmp AS (
+    SELECT
+        td.date_day::date AS activity_date,
+        server_id AS server_id,
+        user_role AS user_role,
+        server_version AS server_version,
+        SUM(CASE WHEN td.date_day::date = event_date THEN promoters ELSE 0 END) AS promoters,
```

Comment:
Not sure why the sum is required here. Data are already aggregated on the previous CTE. Wouldn't a `COALESCE` be enough here?

Reply:
Without the `SUM` the code is creating 2 rows per `activity_date`, 1 row when there are promoters and another extra row with `0`'s. This is causing the quarterly calculation to break in the next CTE as I am using a window function on prior 90 days